### ISSUE TITLE
Deep Links

### ISSUE THREAD

**Issue by nicholasio** [pantheon-systems/wp-saml-auth#101]
It does't sound like this plugin supports deep links (i.e redirecting to the original requested page upon login).

We're using this plugin on a intranet site, and users must be logged in to view the site (it's under Restricted Site Access). The default behavior of the plugin is to just redirect user to home page, no matter what the url it is trying to access. By looking at the plugin code I didn't found a way to make deeps link works without modifying its code. Would you be open for a PR to add support for it? I already have this working on a fork of this plugin, basically this is what I'm doing:

````
$redirect_to = filter_input( INPUT_GET, 'redirect_to', FILTER_SANITIZE_URL );
			
if ( ! $redirect_to ) {
    $redirect_to = '/'
}

$this->provider->requireAuth(
	array(
	   'ReturnTo' => add_query_arg( array( 'redirect_to' => rawurlencode( $redirect_to ), $_SERVER['REQUEST_URI'] ) ),
	)
);
````

redirect_to is being forward to the plugin action's url if one is being set on wp-login.php.

**Comment by danielbachhuber** [comment]
Yep, this would be fine. Let's make sure we're only redirecting to URIs on the same host.

**Comment by danielbachhuber** [comment]
Shipped v0.3.3: https://github.com/pantheon-systems/wp-saml-auth/releases/tag/v0.3.3

Thanks again for your help!
